src/realm/descriptor.hpp

```diff
@@ -293,6 +293,8 @@ class Descriptor : public std::enable_shared_from_this<Descriptor> {
     /// \param link_type The type of links the column should store.
     void set_link_type(size_t col_ndx, LinkType link_type);
 
+    Spec* get_spec() const;
```

Comment:
Unless there is a very good reason, `Descriptor` must not have a public `get_spec()`, because `Descriptor` is part of the public API, but `Spec` is not.